chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -260,7 +260,15 @@ post_state() {
 # -----------------------
 
 MAC=$(cat /sys/class/net/$BOOTDEV/address)
+<% if node[:provisioner][:keep_existing_hostname] -%>
+HOSTNAME=$(cat /etc/HOSTNAME)
+if ! echo "$HOSTNAME" | grep -q -o "$DOMAIN$"; then
+  echo "The fully qualified domain name did not contain the cloud domain."
+  exit 1
+fi
+<% else -%>
 HOSTNAME="d${MAC//:/-}.${DOMAIN}"
+<% end -%>
 sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 $HOSTNAME ${HOSTNAME%%.*} localhost.localdomain localhost/" /etc/hosts
 echo "$HOSTNAME" > /etc/HOSTNAME
 hostname "$HOSTNAME"
```

Comment:
It's unclear to me if we want to do these last 3 lines when we preserve the hostname. I would expect that we wouldn't want that, but I'm not 100% sure.

There's also https://github.com/rsalevsky/barclamp-provisioner/blob/hostname/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb#L329 to consider